### DISCUSSION TITLE
[QUESTION] Is there a purpose in creating nodes for each item added to the inventory?

### DISCUSSION THREAD

**Discussion by scriptsengineer** [peter-kish/gloot#82] [Q&A] [answered]
I believe that inventory only needs to add information about items by slots, creating a node for each item generates a lot of unnecessary information for the godot scene tree

**Comment by peter-kish** [comment, marked as answer]
I made that design decision pretty early during development when the inspector part of the plugin still didn't exist (including the inventory editor, property editor, protoset editor etc.). Having inventory items represented as nodes made populating the inventories and editing item properties from the editor much easier.

Another reason for this design was that it just seemed like the Godot-way of doing composition, as a lot of the engine components are added as child nodes (even ones that don't inherit from `Node3D` or `CanvasItem`).

Now that editing items and inventories using special controls in the inspector is working decently, this design might not make a lot of sense any more. Reverting it could simplify parts of the implementation, but since the current logic revolves around items being nodes I don't expect it to be a simple task. It will definitely take some to analyze the current code and define all the steps required to pull it off and the potential problems.

**Reply by scriptsengineer** [reply]
Thanks for the answer, It makes perfect sense to have something easy to edit, but to approximate the inventory with patterns used in the editor nodes, it can easily encapsulate this item/slot information
